lib/mongo/client.rb

```diff
@@ -902,9 +906,14 @@ def watch(pipeline = [], options = {})
 
     # Create a new encrypter object using the client's auto encryption options
     def build_encrypter
-      @encrypter = Crypt::AutoEncrypter.new(
-        @options[:auto_encryption_options].merge(client: self)
-      )
+      begin
+        @encrypter = Crypt::AutoEncrypter.new(
+          @options[:auto_encryption_options].merge(client: self)
+        )
+      rescue
+        @cluster.disconnect!
```

Comment:
Can you move this to the client constructor please so that the construction and teardown are invoked from the same place?

Reply:
Done!